doc/api/events.md

```diff
@@ -935,13 +935,351 @@ if the `EventEmitter` emits `'error'`. It removes all listeners when
 exiting the loop. The `value` returned by each iteration is an array
 composed of the emitted event arguments.
 
+## `EventTarget` and `Event` API
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+The global `EventTarget` and `Event` objects are a Node.js-specific adaptation
+of the [`EventTarget` Web API][] that are exposed by some Node.js core APIs.
+Neither the `EventTarget` nor `Event` classes are currently available for end
+user code to create.
+
+```js
+const target = getEventTargetSomehow();
+
+target.dispatchEvent({ type: 'foo' });
+
+target.addEventListener('foo', (event) => {
+  console.log('foo event happened!');
+});
+```
+
+The `EventTarget` class is similar in function to `EventEmitter` but evolved
+separately as part of browser DOM APIs. The Node.js implementation of
+`EventTarget` is an adaptation of the DOM API with a number of Node.js-specific
+extensions and behaviors intended to allow it to work more naturally within
+the Node.js environment.
+
+### Node.js `EventTarget` vs. DOM `EventTarget`
+
+There are four key differences between the Node.js `EventTarget` and the
+[`EventTarget` Web API][]:
+
+1. There is no concept of a hierarchical DOM and event propagation in Node.js.
+   That is, an event dispatched to an `EventTarget` does not propagate through
+   a hierarchy of nested target objects that may each have their own set of
+   handlers for the event.
+2. In the Node.js `EventTarget`, any object with a `type` property whose value
+   is a string may be dispatched as an `event`.
+3. In the Node.js `EventTarget`, if an event listener is an async function
+   or returns a `Promise`, and returned `Promise` rejects, the rejection will
+   be automatically captured and handled the same way as a listener that throws
+   synchronously (see [`EventTarget` Error Handling][] for details).
+4. The Node.js `EventTarget` implements an additional set of APIs that allow it
+   to closely emulate a subset of the `EventEmitter` API.
```

Comment:
I would suggest putting those outside the class, e.g. as module-level, so as to make it clear when people are writing non-interoperable code (or let them make it interoperable via browserify-type shims of the module exports).

Reply:
As far as I can see, there is nothing in the spec that prevents `EventTarget` from being extended. If anything here, we can expose two interfaces: a base `EventTarget` that exposes the subset of `EventTarget` APIs that make sense in Node.js and an `EmitterEventTarget` that extends that to add the additional `EventEmitter` emulation.

Reply:
@jasnell exposing two APIs makes sense to me, I also suggested it above. Sounds like that would be a good way to resolve a divergence.

Reply:
> As far as I can see, there is nothing in the spec that prevents EventTarget from being extended

By "extended" do you mean "subclassed", or "implemented with additional methods"? The latter is definitely prevented, by virtue of the fact that the class definition does not contain any additional methods. (Plus the Web IDL spec, which specifies how classes are created from interface definitions.)